File: Networking/src/main/java/jsonprotocol/JsonProtocolUtils.java
package jsonprotocol;


import org.model.Cursa;
import org.model.Participant;
import org.model.User;

import java.util.List;

public class JsonProtocolUtils {
    public static Request createLoginRequest(User user){
        Request req=new Request();
        req.setType(RequestType.LOGIN);
        req.setUser(user);
        return req;
    }

    public static Response createUserResponse(User user){
        Response res=new Response();
        res.setType(ResponseType.USER_EXISTS);
        res.setUser(user);
        return res;
    }

    public static Response createOkResponse(){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        return resp;
    }

    public static Response createErrorResponse(String errorMessage){
        Response resp=new Response();
        resp.setType(ResponseType.ERROR);
        resp.setErrorMessage(errorMessage);
        return resp;
    }
    public static Request createGetAllTeamsRequest(){
        Request req=new Request();
        req.setType(RequestType.GET_ALL_TEAMS);
        return req;
    }

    public static Response createGetTeamResponse(List<String> teams){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        resp.setEchipe(teams);
        return resp;
    }

    public static Request createRequestGetCapacitati(){
        Request req=new Request();
        req.setType(RequestType.GET_CAPACITATI);
        return req;
    }
    public static Request createRequestGetAllParticipanti(){
        Request req=new Request();
        req.setType(RequestType.GET_ALL_PARTICIPANTI);
        return req;
    }
    public static Request createRequestGetAllCurse(){
        Request req=new Request();
        req.setType(RequestType.GET_ALL_CURSE);
        return req;
    }

    public static Request createRequestGetCurseByCap(int cap){
        Request req=new Request();
        req.setType(RequestType.GET_CURSE_BY_CAP);
        req.setCapMotor(cap);
        return req;
    }

    public static Request createRequestGetParticipantiByEchipa(String echipa){
        Request req=new Request();
        req.setType(RequestType.GET_PARTICIPANTI_BY_ECHIPE);
        req.setEchipa(echipa);
        return req;
    }

    public static Response createGetCapacitatiResponse(List<Integer> cap){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        resp.setCapacitatiMotor(cap);
        return resp;
    }

    public static Response createParticipantiResponse(List<Participant> par){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        resp.setParticipanti(par);
        return resp;
    }

    public static Response createCurseResponse(List<Cursa> curse){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        resp.setCurse(curse);
        return resp;
    }

}

File: networking/jsonprotocol/JsonProtocolUtils.cs
using model.ORMModel;

namespace networking.jsonprotocol;

public class JsonProtocolUtils {
    public static Response CreateOkResponse()
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        return resp;
    }

    public static Request CreateLoginRequest(User user)
    {
        Request req = new Request();
        req.Type = RequestType.LOGIN;
        req.User = user;
        return req;
    }

    public static Response CreateUserResponse(User user)
    {
        Response res = new Response();
        res.Type = ResponseType.OK;
        res.User = user;
        return res;
    }

    public static Response CreateErrorResponse(string errorMessage)
    {
        Response resp = new Response();
        resp.Type = ResponseType.ERROR;
        resp.ErrorMessage = errorMessage;
        return resp;
    }

    public static Request CreateLogOutRequest(User user)
    {
        Request req = new Request();
        req.Type = RequestType.LOGOUT;
        req.User = user;
        return req;
    }

    public static Request CreateAdaugaParticipantRequest(Participant participant)
    {
        Request req = new Request();
        req.Type = RequestType.ADAUGA_PARTICIPANT;
        req.Participant = participant;
        return req;
    }

    public static Request CreateAdaugaInscriereRequest(Participant participant, Cursa cursa)
    {
        Request req = new Request();
        req.Type = RequestType.ADAUGA_INSCRIERE;
        req.Participant = participant;
        req.Cursa = cursa;
        return req;
    }

    public static Request CreateAdaugaUserRequest(User user)
    {
        Request req = new Request();
        req.Type = RequestType.ADAUGA_USER;
        req.User = user;
        return req;
    }

    public static Response CreateUpdateResponse()
    {
        Response res = new Response();
        res.Type = ResponseType.UPDATE;
        return res;
    }

    public static Request CreateRequestGetCurseByParticipant(Participant participant)
    {
        Request req = new Request();
        req.Type = RequestType.GET_CURSE_BY_PARTICIPANT;
        req.Participant = participant;
        return req;
    }

    public static Request CreateGetAllTeamsRequest()
    {
        Request req = new Request();
        req.Type = RequestType.GET_ALL_TEAMS;
        return req;
    }

    public static Response CreateGetTeamResponse(List<string> teams)
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        resp.Echipe = teams;
        return resp;
    }

    public static Request CreateRequestGetCapacitati()
    {
        Request req = new Request();
        req.Type = RequestType.GET_CAPACITATI;
        return req;
    }

    public static Request CreateRequestGetAllParticipanti()
    {
        Request req = new Request();
        req.Type = RequestType.GET_ALL_PARTICIPANTI;
        return req;
    }

    public static Request CreateRequestGetAllCurse()
    {
        Request req = new Request();
        req.Type = RequestType.GET_ALL_CURSE;
        return req;
    }

    public static Request CreateRequestGetCurseByCap(int cap)
    {
        Request req = new Request();
        req.Type = RequestType.GET_CURSE_BY_CAP;
        req.CapMotor = cap;
        return req;
    }

    public static Request CreateRequestGetParticipantiByEchipa(string echipa)
    {
        Request req = new Request();
        req.Type = RequestType.GET_PARTICIPANTI_BY_ECHIPE;
        req.Echipa = echipa;
        return req;
    }

    public static Response CreateGetCapacitatiResponse(List<int> capacitati)
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        resp.CapacitatiMotor = capacitati;
        return resp;
    }

    public static Response CreateParticipantiResponse(List<Participant> participants)
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        resp.Participanti = participants;
        return resp;
    }

    public static Response CreateCurseResponse(List<Cursa> curse)
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        resp.Curse = curse;
        return resp;
    }

}
